cpp/modules/deck.gl/core/src/viewports/web-mercator-viewport.cpp

```diff
@@ -98,7 +98,59 @@ auto WebMercatorViewport::getMapCenterByLngLatPosition(mathgl::Vector2<double> l
   return this->unprojectFlat(newCenter);
 }
 
-auto fitBounds(mathgl::Vector2<double> topLeft, mathgl::Vector2<double> bottomRight, int padding = 0,
-               mathgl::Vector2<int> offset = mathgl::Vector2<int>()) {
-  throw new std::logic_error("fitBounds not supported");
+auto WebMercatorViewport::fitBounds(mathgl::Vector2<double> topLeft, mathgl::Vector2<double> bottomRight, int padding,
+                                    mathgl::Vector2<int> offset) -> WebMercatorViewport {
+  // TODO(randy@unfolded.ai)
+  // https://github.com/uber-web/math.gl/blob/master/modules/web-mercator/src/fit-bounds.js
+
+  WebMercatorViewport::Options options;
+  options.width = this->width;
+  options.height = this->height;
+  options.longitude = 0;
+  options.latitude = 0;
+  options.zoom = 0;
+
+  // Assume constant padding for now
+  // Call WebMercatorViewport constructor, passing in opts
+  WebMercatorViewport viewport = WebMercatorViewport(options);
+
+  auto nw = viewport.projectFlat(topLeft);
+  auto se = viewport.projectFlat(bottomRight);
+
+  // get abs of x/y difference
+  // max(abs, minExtent)?
+  const double size[] = {abs(se.x - nw.x), abs(se.y - nw.y)};
+
+  const double targetSize[] = {this->width - 2 * padding - abs(offset.x) * 2,
```

Comment:
I'd suggest making these `Vector2<double>`s. The JavaScript deck.gl uses arrays as its vector type.